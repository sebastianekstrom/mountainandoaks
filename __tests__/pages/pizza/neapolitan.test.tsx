import { render, screen } from "@testing-library/react";
import React from "react";

import Neapolitan from "pages/pizza/neapolitan";

describe("Neapolitan page", () => {
  it("renders title correctly", () => {
    render(<Neapolitan />);
    const title = screen.queryAllByText(/Neapolitan style pizza dough/i);
    expect(title).toBeDefined();
  });

  it("renders the ingress text", () => {
    render(<Neapolitan />);
    const ingress = screen.queryAllByText(/The classic of the classics/i);
    expect(ingress).toBeDefined();
  });

  it("renders the correct ingredients", () => {
    render(<Neapolitan />);
    const ingredientList = screen.getByTestId("INGREDIENT_LIST");
    expect(ingredientList).toMatchSnapshot();
  });

  it("renders the method steps", () => {
    render(<Neapolitan />);
    const step = screen.queryAllByText(
      /Transfer the dough to a lightly oiled bowl/i,
    );
    expect(step).toBeDefined();
  });

  it("renders the alert with the correct description", () => {
    render(<Neapolitan />);
    const alert = screen.queryAllByText(/For optimal results I recommend/i);
    expect(alert).toBeDefined();
  });
});
