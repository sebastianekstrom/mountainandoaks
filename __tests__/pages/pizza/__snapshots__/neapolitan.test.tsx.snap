// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Neapolitan page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="flex flex-col"
  >
    <ul>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Flour (Tipo 00)
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          166
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Water
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          103
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Salt
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          5
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Dry yeast
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          1
          g
        </p>
      </li>
    </ul>
    <div
      class="ml-auto"
    >
      <button
        class="mt-2 rounded-full text-regular bg-pill-background py-2 px-4"
        data-testid="BAKERS_PERCENTAGE"
        role="button"
        type="button"
      >
        <p
          class="variable font-sans text-caption leading-paragraph "
        >
          Show bakers %
        </p>
      </button>
    </div>
  </div>
</section>
`;
